test/src/test/java/org/corfudb/integration/CorfuReplicationLargeTxIT.java

```diff
@@ -110,22 +114,38 @@ private void testTxChunking(int numEntriesToWrite,
         corfuStoreStandby.subscribeListener(standbyListener, LogReplicationMetadataManager.NAMESPACE,
                 LR_STATUS_STREAM_TAG);
 
-        // Calculate the expected total number of streaming updates across all
-        // tables
+        // Calculate the expected total number of streaming updates across all tables
         int totalStreamingUpdates =
             expectedStreamingUpdatesPerTable * mapNameToMapStandby.size();
 
+        // MAX_SNAPSHOT_ENTRIES_APPLIED = 1.  So entries to the ProtobufDescriptor table will be applied in a batch
+        // of 1.  Hence, the number of transactions made = number of entries received from the Source Cluster(1 TX
+        // per entry).
+        int numExpectedTxOnProtobufDescriptorTable = activeRuntime.getTableRegistry().getProtobufDescriptorTable().size();
+
+        // The number of entries in the protobuf descriptor table on Source(or numExpectedTxOnProtobufDescriptorTable)
+        // must not be equal to MAX_SNAPSHOT_ENTRIES_APPLIED.  Otherwise we cannot verify that it was applied in
+        // batches.
+        Assert.assertNotEquals(numExpectedTxOnProtobufDescriptorTable, MAX_SNAPSHOT_ENTRIES_APPLIED);
```

Comment:
Just curious what is the real value if not MAX_SNAPSHOT_ENTRIES_APPLIED here in this IT?
Wondering if it is feasible to create a test case where a large number of updates causes the write size case to trigger forcing multiple updates made to the protobuf descriptor table even though on the source there is only one?

Reply:
MAX_SNAPSHOT_ENTRIES_APPLIED in this test is 1.  The protobuf descriptor table will definitely have > 1 entries as we are opening the data tables.  So the case of chunking it is exercised here.

Regarding the other test which forces multiple updates on the proto table even if the source has only 1 entry, it would have to exercise the upgrade code path by changing the version in the protobuf descriptor table.